app/services/calendar_factory_from_event_hash.rb

```diff
@@ -24,20 +26,19 @@ def create
 
 private
 
+  def parent_calendar
+    Calendar.find_by(calendar_area: @area, template: true) || Calendar.find_by(calendar_area: @area.parent_area, template: true)
+  end
+
   def create_holidays_between_terms
     HolidayFactory.new(@calendar).create
   end
 
   def create_bank_holidays
-    calendar_event_type = CalendarEventType.bank_holiday.first
-    find_bank_holidays(@area).each do |bh|
-      @calendar.calendar_events.where(title: bh.title, start_date: bh.holiday_date, end_date: bh.holiday_date, calendar_event_type: calendar_event_type).first_or_create!
-    end
-  end
+    bank_holiday = CalendarEventType.bank_holiday.first
```

Comment:
If we do what I'm suggesting above then we can copy across all calendar events from the template calendar here. Seems odd that we've got a template but we're only copying across one type of event.

Reply:
This can now just be, copy everything from parent calendar (Which will have the BHs already)